helpers/error_handler/error_message.go

```diff
@@ -51,6 +53,8 @@ func DatabaseErrorMessage(err error) error {
 			return errors.New(somethingWentWrongError)
 		case duplicateErrorCode:
 			return errors.New(emailDuplicateError)
+		case invalidInputErrorCode:
```

Comment:
Instead of checking on `pgerr.code`, it migth be better to use `pgerr.Code.Class().Name()`:

```go
switch pgerr.Code.Class().Name() {
case "data_exception":
  err = ErrInvalidValue
case "integrity_constraint_violation":
  // This could be lots of things: foreign key violation,
  // non-null constraint violation, etc., but we're generally
  // checking those in advance. As long as our code is in
  // order, unique constraints will be the only things we're
  // actually relying on the database to check for us.
  err = ErrDuplicate
}
```

Here is the list of values: https://github.com/lib/pq/blob/master/error.go#L78

You can see more info on [this blog post](https://ders.github.io/post/2016-12-23-lean-clean-golang-machine/).

What do you think?

Also, there seem to be an attribute `Message` and `Detail`:

https://github.com/lib/pq/blob/4604d39ddc9f62e2ca152114c73aec99b77a3468/error.go#L28-L29

So wondering if this could be used to have better error messages too.

Reply:
`GORM` doesn't use `https://github.com/lib/pq` but they use `github.com/jackc/pgconn` so `Class()` and `Name()` functions are not available.

pgconn also provide `Message` and `Detail` attributes and this is the result
- Message : `invalid input syntax for type bigint: "invalidKeyword"`
- Detail does not provide anything

for more information [click](https://github.com/jackc/pgconn/blob/master/errors.go)

Reply:
Thank you for the detailed. Still not helping us to get better error messages 😅 but now I have more info on my hand.